File: src/OrderService/Models/Order.cs
namespace OrderService.Models;

public class Order
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public string Status { get; set; } = "Pending";
    public decimal Total { get; set; }
    public List<OrderItem> Items { get; set; } = new();
}

public class OrderItem
{
    public int Id { get; set; }
    public int OrderId { get; set; }
    public int ProductId { get; set; }
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
}

File: src/frontend/distributed-order-app/src/app/models/order.ts
export interface OrderItem {
  productId: number;
  quantity: number;
  unitPrice: number;
}

export interface CreateOrderRequest {
  items: OrderItem[];
}

export interface CreateOrderResponse {
  orderId: number;
  status: string;
  total: number;
}

export interface OrderItemDetail {
  id: number;
  productId: number;
  quantity: number;
  unitPrice: number;
}

export interface Order {
  id: number;
  createdAt: string;
  status: string;
  total: number;
  items: OrderItemDetail[];
}
